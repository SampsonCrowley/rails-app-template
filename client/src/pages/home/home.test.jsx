import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Home from './home';

describe('Pages - Home', () => {
  const div = document.createElement('div');
  const createHome = ({...props}) => {
    ReactDOM.render(<Home {...props} />, div);
    return div.querySelector('div.Site-intro')
  }

  it('renders an div with class "Site-intro"', () => {
    const rendered = createHome()
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("DIV")
    ReactDOM.unmountComponentAtNode(div);
  })

  it('sets an unnecessarily tall height', () => {
    const rendered = renderer
      .create(<Home />)
      .toJSON();
    expect(rendered.props.style)
    expect(rendered.props.style.height).toBe('1000vh')
  })

  it('is snapshotable', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot()
  })
})
