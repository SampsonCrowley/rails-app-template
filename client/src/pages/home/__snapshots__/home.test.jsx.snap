// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages - Home is snapshotable 1`] = `
<div
  className="Site-intro"
  style={
    Object {
      "height": "1000vh",
    }
  }
>
  <div
    className="la-ball-atom la-vh"
    style={
      Object {
        "color": "#00F",
        "marginLeft": "auto",
        "marginRight": "auto",
        "zIndex": 0,
      }
    }
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
